R/lazy-join-query.R

```diff
@@ -199,15 +199,26 @@ sql_build.lazy_multi_join_query <- function(op, con, ...) {
 }
 
 generate_join_table_names <- function(table_names) {
+  table_name_length_max <- dplyr::coalesce(max(nchar(table_names$name)), 0)
+
   if (length(table_names$name) != 2) {
     table_names_repaired <- vctrs::vec_as_names(table_names$name, repair = "unique", quiet = TRUE)
     auto_name <- table_names$from != "as"
     table_names$name[auto_name] <- table_names_repaired[auto_name]
-
-    return(table_names$name)
+    table_names_prepared <- table_names$name
+  } else{
+    table_names_prepared <- join_two_table_alias(table_names$name, table_names$from)
   }
 
-  join_two_table_alias(table_names$name, table_names$from)
+  # avoid database aliases exceeding the database-specific maximum length
+  abbreviate(
+    table_names_prepared,
+    # arbitrarily floor at 63 (Postgres limit) to avoid unnecessarily truncating reasonable lengths
+    minlength = max(table_name_length_max, 63),
+    # (explicit, default) err on the side of unique table names instead of exact length
```

Comment:
I don't understand what this comment is saying. Can you clarify?

Reply:
There is seemingly a fundamental limitation to `abbreviate()` which does not allow for both (a) max length to be binding and (b) names to be guarenteed unique.

From the documentation:
```
strict | logical: should minlength be observed strictly? Note that setting strict = TRUE may return non-unique strings.
```

I've opted for `strict = FALSE` to preserve uniqueness (since `dbplyr` will require it any), but there is a non-zero risk that aliases still exceed the database limit. As one mitigation, I've updated the `method` to `both.sides` which tends to add a bit more scrambling (slightly less obvious alias names) but the intended length is more likely to be aligned with the requested length.

I've spent a decent amount of time playing with combinations of args (including discussing with ChatGPT) but couldn't come up with a fully-error proof solution using `abbreviate` _or_ an alternate function.